### PR TITLE
Add rho related checks

### LINKED ISSUE #103: [Keccak] Implement Rho and Pi circuit
Pi is trivial, should be resolved in 1 or 2 rows.

Rho needs these checks

- Running sum by slices
- block count checks

Some initial work is in #54

### PR DESCRIPTION
**Pull request by ChihChengLiang**
Fix  #103

### CODE REVIEW

**Review by CPerezz — APPROVED** [review]
LGTM. Small nits.

File: keccak256/src/gates/rho_checks.rs [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+            if is_at_rotation_offset(new.chunk_idx, self.rotation) {
+                BigUint::one()
+            } else {
+                self.output_power_of_base.clone() * B9.pow(self.step)
+            };
+        new.output_acc +=
+            self.output_power_of_base.clone() * self.output_coef.clone();
+        // Case of last chunk, aka special chunks
+        if new.chunk_idx >= LANE_SIZE {
+            assert_eq!(new.input_raw, BigUint::zero());
```
[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply keccak256/src/gates/rho_checks.rs:139]: This is the place I wish it to panic if that invariant doesn't hold. `debug_assert!` seems not compiled in the production.

File: keccak256/src/gates/rho_checks.rs [hunk 2]
```diff
@@ -0,0 +1,709 @@
...
+            new.block_count = None;
+            new.high_value = Some(biguint_mod(&self.input_raw, B13));
+            new.output_coef = BigUint::from(convert_b13_coef(
+                new.high_value.unwrap() + self.low_value,
+            ));
+            assert_eq!(
```
[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply keccak256/src/gates/rho_checks.rs:145]: ditto for same reason

File: keccak256/src/gates/rho_checks.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (CPerezz) [reply keccak256/src/gates/rho_checks.rs:139]: Then we should maybe throw an error or panic with some message explaining briefly the reason why we panic no? 
Otherways, the panic is a bit out of any explanation and simply happens.

[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply keccak256/src/gates/rho_checks.rs:139]: Sure, will add

[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply keccak256/src/gates/rho_checks.rs]: added the reasons to panic in 5b735eb

**Review by CPerezz — APPROVED** [review]
Looks really good! 

The documentation is really good and makes it much easier to understand! Thanks a lot for the awesome work @ChihChengLiang !!! :heart:

**Pull request merged**